frontend/src/pages/Main.tsx

```diff
@@ -185,6 +186,97 @@ const Main: FC = () => {
     });
   }, []);
 
+  /**
+    /* Function to calculate the maximum distance
+    /*
+    /* @author Chen
+    */
+  const calculateMaxDistance = (selectedRoad: IRoad): number => {
+    const wayIds = Object.keys(selectedRoad.geometries);
+    let maxDistance = 0;
+    let farthestCoordinate: L.LatLng | null = null;
+
+    // Convert all coordinates to Leaflet LatLng objects and store in an array
+    const allLeafletCoordinates: L.LatLng[] = wayIds.flatMap((wayId) =>
+      selectedRoad.geometries[wayId].map(
+        (coord) => L.latLng(coord.lat, coord.lng), // Use the L.latLng function
+      ),
+    );
+    // If there are no coordinates, exit early
+    if (allLeafletCoordinates.length === 0) {
+      return 0;
+    }
+    // Initialize farthestCoordinate with the first coordinate, if available
+    if (allLeafletCoordinates.length > 0) {
+      farthestCoordinate = allLeafletCoordinates[0];
+    }
+    // Find the point farthest from the first coordinate
+    const firstLeafletCoordinate = allLeafletCoordinates[0];
+    allLeafletCoordinates.forEach((leafletCoordinate) => {
+      const distance = firstLeafletCoordinate.distanceTo(leafletCoordinate);
+      if (distance > maxDistance) {
+        maxDistance = distance;
+        farthestCoordinate = leafletCoordinate;
+      }
+    });
+
+    // Reset max distance for the second iteration
+    maxDistance = 0;
+
+    // Find the point farthest from the farthest point found in step 1
+    if (farthestCoordinate) {
+      allLeafletCoordinates.forEach((leafletCoordinate) => {
+        const distance = farthestCoordinate!.distanceTo(leafletCoordinate);
+        if (distance > maxDistance) {
+          maxDistance = distance;
+        }
+      });
+    }
+
+    return maxDistance;
+  };
+
+  /**
+    /* Function to calculate the zoom level
+    /*
+    /* @author Chen
+    */
+  const calculateZoomLevel = (maxDistance: number): number => {
+    let slope = -7.8e-5; // Default
+    let intercept = 12.6; // Default
+
+    if (maxDistance <= 900) {
+      slope = -7.8e-5;
+      intercept = 16.8;
+    } else if (Math.floor(maxDistance) == 50937) {
+      return 13; //For a road named 'Nordre Ringvej' that cannot get the correct MaxDistance
```

Comment:
I'm not very comfortable having edge cases in that kind of code... Maybe using the length of the road would allow not to have edge cases in the code?